### ISSUE TITLE
Localstack Pro + localstack-chalice + AWS Chalice not working

### ISSUE THREAD

**Issue by tyagow** [localstack/localstack#3260]
[x] bug report
[ ] feature request

# Detailed description
Using localstack-chalice with was chalice is not working.
I thought the error was because Chalice uses Lambda layers which is a localstack pro feature, but after subscribing and adding the KEY I still get the same error.

## Expected behavior
run `chalice-local deploy local` and get all chalice resources deployed to localstack

## Actual behavior
When I do `chalice-local deploy local` I get the following:

```
chalice.deploy.deployer.ChaliceDeploymentError: ERROR - While deploying your chalice application, received the following error:

 An error occurred (405) when calling the DeleteFunctionConcurrency operation: 
 <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>405 Method Not 
 Allowed</title>
<h1>Method Not Allowed</h1>
<p>The method is not allowed for 
 the requested URL.</p>
```

my `.chalice/config-local.json`

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": ["ssm:GetParameter"],
      "Resource": ["arn:aws:ssm:*:*:parameter/REDACTED/local/*"],
      "Effect": "Allow"
    },
    {
      "Action": [
        "logs:CreateLogGroup",
        "logs:CreateLogStream",
        "logs:PutLogEvents"
      ],
      "Resource": "arn:aws:logs:*:*:*",
      "Effect": "Allow"
    }
  ]
}
```

Part of my app.py (only the functions)

```

@app.on_sqs_message(queue='my-queue', batch_size=1)
def handle_sqs_message(event):
    print("event", event)
    for record in event:
        app.log.debug("Received message with contents: %s", record.body)


@app.route('/job-worker', methods=['POST'])
def job_worker():
    body = app.current_request.json_body
    print("request.post", body)
```

After I run  `chalice-local deploy local` I check localstack and there just one function created there:

```
╰─± aws --endpoint-url=http://0.0.0.0:4566 lambda list-functions
{
    "Functions": [
        {
            "FunctionName": "ftg_chalice-local-handle_sqs_message",
            "FunctionArn": "arn:aws:lambda:us-east-1:000000000000:function:ftg_chalice-local-handle_sqs_message",
            "Runtime": "python3.8",
            "Role": "arn:aws:iam::000000000000:role/ftg_chalice-local-handle_sqs_message",
            "Handler": "app.handle_sqs_message",
            "CodeSize": 7452977,
            "Description": "",
            "Timeout": 60,
            "MemorySize": 128,
            "LastModified": "2020-11-20T18:11:58.196+0000",
            "CodeSha256": "JidgtUqdyNwEQPlXX6evd3YSoq23JAjZ18qg843r14k=",
            "Version": "$LATEST",
            "VpcConfig": {
                "SubnetIds": [],
                "SecurityGroupIds": []
            },
            "Environment": {
                "Variables": {
                    "ENV": "local"
                }
            },
            "TracingConfig": {
                "Mode": "PassThrough"
            },
            "RevisionId": "336fffb3-940f-4901-803d-929d0878387d",
            "Layers": [],
            "State": "Active",
            "LastUpdateStatus": "Successful"
        }
    ]
}

```

localstack logs
```
2020-11-20T18:11:58:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:us-east-1:000000000000:function:ftg_chalice-local-handle_sqs_message
```

# Steps to reproduce
```
╰─± SERVICES=sqs,sts,iam,cloudformation,lambda,apigateway,ssm DATA_DIR=/tmp/localstack/data PORT_WEB_UI=8081 DEBUG=1 LOCALSTACK_API_KEY=REDACTED localstack start
```

have a chalice app with the configuration above and run `chalice-local deploy local`

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-337) by [Unito](https://www.unito.io/learn-more)

**Comment by tyagow** [comment]
Maybe this information from chalice developers can help ->
https://github.com/aws/chalice/issues/1585#issuecomment-732299004

**Comment by tyagow** [comment]
Based on the PR open I guess you are already fixing it, thank you very much!!

**Comment by whummer** [comment]
Thanks for reporting @tyagow - this should be fixed in #3269. Can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks for your help.

**Comment by tyagow** [comment]
Hi @whummer thanks you for the fix, I tested with the latest docker image, and indeed the error is solved, but now I faced another API not implemented by localstack more info here - https://github.com/localstack/localstack/issues/3261
thank you for the help
